include/merlin/types.cuh

```diff
@@ -56,7 +56,10 @@ using AtomicPos = cuda::atomic<T, cuda::thread_scope_device>;
 template <class K, class V, class S>
 struct Bucket {
   AtomicKey<K>* keys_;
+  /// TODO: compute the pointer of scores and digests using bucket_max_size
   AtomicScore<S>* scores_;
+  /// @brief not visible to users
+  uint8_t* digests;
```

Comment:
Inspired by a discussion with you, potentially, we can reduce the memory consumption of `Bucket` struct by canceling the separate pointers for `keys_`, `scores_`, and `digests_`, because we just need only 1 start pointer for these three.  
So could you switch the `digests` to a function like this?
```cpp
  __forceinline__ __device__ uint8_t* digests(int index) const {
    return digests_ + index;
  }
```
This will benefit the future refactoring in the future I said.